support/cas-server-support-pac4j-webflow/src/main/java/org/apereo/cas/web/flow/DelegatedClientAuthenticationAction.java

```diff
@@ -202,8 +205,12 @@ protected void prepareForLoginPage(final RequestContext context) {
         final String name = client.getName();
         final Matcher matcher = PAC4J_CLIENT_SUFFIX_PATTERN.matcher(client.getClass().getSimpleName());
         final String type = matcher.replaceAll(StringUtils.EMPTY).toLowerCase();
-        final String redirectUrl = DelegatedClientNavigationController.ENDPOINT_REDIRECT
-            + "?" + Pac4jConstants.DEFAULT_CLIENT_NAME_PARAMETER + "=" + name;
+        val uriBuilder = UriComponentsBuilder.fromUriString(DelegatedClientNavigationController.ENDPOINT_REDIRECT).queryParam(Pac4jConstants.DEFAULT_CLIENT_NAME_PARAMETER, name);
+        val serviceParam = webContext.getRequestParameter(CasProtocolConstants.PARAMETER_SERVICE);
```

Comment:
how would this work with the calling client using SAML1 instead of CAS?

Reply:
While this is a partial solution as is, we would need to investigate use of SAML and cannot merge this as is. I recommend you spend a bit more time on this, to stand up a simple java client using SAML1, and get it to use delegation as well to see how the change behaves. 

I also dont think you need to extract the service parameter manually. There are argument-extractors and such for that already that should handle the saml bit too.

Reply:
This should get you going: https://github.com/cas-projects/cas-sample-java-webapp

And it would be even better to update tests to account for the changes you have added.